### PR TITLE
Add mrope op fusion

### PR DESCRIPTION
**Pull request by shaopeng-666**
### What this PR does / why we need it?
Add mrope fusion op for qwen2.5-vl. This mrope operator dosen't support Qwen3-VL currently. Thus could only take affect in qwen2.5-vl
### Does this PR introduce _any_ user-facing change?
no
### How was this patch tested?

- vLLM version: v0.11.0rc3
- vLLM main: https://github.com/vllm-project/vllm/commit/v0.11.0

### CODE REVIEW

File: vllm_ascend/ops/rotary_embedding.py [hunk 1]
```diff
@@ -395,3 +395,34 @@ def forward(self,
...
+        positions: torch.Tensor,
+        query: torch.Tensor,
+        key: torch.Tensor,
+    ):
+        import torch_npu
+        mrope_section = [0, 0, 0
+                         ] if positions.ndim == 1 else self.mrope_section
+
+        if self.cos_sin_cache.device != query.device:  # type: ignore
+            self.cos_sin_cache = self.cos_sin_cache.to(
```
Comment (MengqingCao) [inline vllm_ascend/ops/rotary_embedding.py]:
```suggestion
            self.cos_sin_cache = self.cos_sin_cache.to(  # type: ignore
```

Comment (shaopeng-666) [reply vllm_ascend/ops/rotary_embedding.py]: fixed

File: vllm_ascend/ops/rotary_embedding.py [hunk 2]
```diff
@@ -395,3 +395,34 @@ def forward(self,
...
+
+        if self.cos_sin_cache.device != query.device:  # type: ignore
+            self.cos_sin_cache = self.cos_sin_cache.to(  # type: ignore
+                query.device)  # type: ignore
+
+        if self.cos_sin_cache.dtype != query.dtype:  # type: ignore
+            self.cos_sin_cache = self.cos_sin_cache.to(  # type: ignore
+                query.dtype)  # type: ignore
+
+        query, key = torch_npu.npu_mrope(positions,
```
Comment (MengqingCao) [inline vllm_ascend/ops/rotary_embedding.py:423]: Plz make sure the accuracy issue of this operator is ok

**Comment by MengqingCao** [comment]
plz update the pr message, and will this custom op of mrope will bring performance gain?

File: vllm_ascend/ops/rotary_embedding.py (hunk 2 above)
Comment (shaopeng-666) [reply vllm_ascend/ops/rotary_embedding.py:423]: This fusion op only support Qwen2.5-VL now, have added op register judge in utils.py

**Review by MengqingCao — APPROVED** [review]
LGTM
